src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -842,6 +842,6 @@ def run_radosgw(self, port=None, log_file=None, debug_ms=None, debug_rgw=None):
         if debug_rgw:
             params += ['--debug-rgw', debug_rgw]
 
-        (retcode, stdout, stderr) = RGWCmd(self.env).run(params)
+        (stdout, stderr) = RGWCmd(self.env).run(params)
 
         return (retcode, stdout, stderr)
```

Comment:
The make check failure is related. If you're going to remove `retcode` from line 845, you also have to remove it in the return statement.
```
py3 run-test: commands[2] | mypy --config-file=../mypy.ini -p ceph
Success: no issues found in 27 source files
lint create: /home/jenkins-build/build/workspace/ceph-pull-requests/src/python-common/.tox/lint
lint installdeps: -rrequirements-lint.txt
WARNING: Discarding $PYTHONPATH from environment, to override specify PYTHONPATH in 'passenv' in your configuration.
lint inst: /home/jenkins-build/build/workspace/ceph-pull-requests/src/python-common/.tox/.tmp/package/1/ceph-1.0.0.zip
WARNING: Discarding $PYTHONPATH from environment, to override specify PYTHONPATH in 'passenv' in your configuration.
lint installed: ceph @ file:///home/jenkins-build/build/workspace/ceph-pull-requests/src/python-common/.tox/.tmp/package/1/ceph-1.0.0.zip,docutils==0.18.1,entrypoints==0.3,flake8==3.7.8,mccabe==0.6.1,pycodestyle==2.5.0,pyflakes==2.1.1,PyYAML==6.0,rstcheck==3.3.1
lint run-test-pre: PYTHONHASHSEED='3224803879'
lint run-test: commands[0] | flake8 ceph
ceph/rgw/rgwam_core.py:847:17: F821 undefined name 'retcode'
ERROR: InvocationError for command /home/jenkins-build/build/workspace/ceph-pull-requests/src/python-common/.tox/lint/bin/flake8 ceph (exited with code 1)
```

Reply:
Indeed, sorry about that. Hopefully this won't be propagated higher up in function call trees. I'll have to check that.

Reply:
Upper in existing call trees, `ZoneCommand.run()` will now return `(stdout, stderr)` instead of `(retcode, stdout, stderr)` too. Shouldn't be an issue, the class is not documented, and the method has obviously never been used. - or it would have failed.

Reply:
I had a look at the functions that return a `stdout, stderr` pair overall ceph. There has been no effort to have consistent return values in the API, but most functions return either `returncode, stdout, stderr` or `stdout, stderr, returncode`:

<details>
 <summary>Functions that return a (stdout, stderr) pair</summary>

```
$ grep -R 'stdout, stderr' --exclude-dir=qa | grep return
doc/mgr/modules.rst:Each command is expected to return a tuple ``(retval, stdout, stderr)``.
src/ceph-volume/ceph_volume/tests/test_process.py:            communicate=lambda x: (stdout, stderr, returncode)
src/ceph-volume/ceph_volume/util/prepare.py:    stdout, stderr, returncode = process.call(
src/ceph-volume/ceph_volume/util/prepare.py:    stdout, stderr, returncode = process.call(
src/ceph-volume/ceph_volume/util/prepare.py:    stdout, stderr, returncode = process.call(
src/ceph-volume/ceph_volume/util/encryption.py:    stdout, stderr, returncode = process.call(
src/ceph-volume/ceph_volume/process.py:    * returns stdout, stderr, and exit code (vs. just the exit code)
src/ceph-volume/ceph_volume/process.py:    return stdout, stderr, returncode
src/ceph-volume/ceph_volume/api/lvm.py:    stdout, stderr, returncode = process.call(args, run_on_host=True, verbose_on_failure=False)
src/ceph-volume/ceph_volume/api/lvm.py:    stdout, stderr, returncode = process.call(args, run_on_host=True, verbose_on_failure=False)
src/ceph-volume/ceph_volume/api/lvm.py:    stdout, stderr, returncode = process.call(
src/ceph-volume/ceph_volume/api/lvm.py:    stdout, stderr, returncode = process.call(
src/ceph-volume/ceph_volume/api/lvm.py:    stdout, stderr, returncode = process.call(args, run_on_host=True, verbose_on_failure=False)
src/ceph-volume/ceph_volume/api/lvm.py:    stdout, stderr, returncode = process.call(
src/ceph-volume/ceph_volume/api/lvm.py:    stdout, stderr, returncode = process.call(
src/ceph-volume/ceph_volume/api/lvm.py:    stdout, stderr, returncode = process.call(
src/cephadm/cephadm:            return stdout, stderr, returncode
src/cephadm/cephadm:    stdout, stderr, returncode = async_run(run_with_timeout())
src/cephadm/cephadm:    return stdout, stderr, returncode
src/rgw/rgwam.py:        return run_cmd, result.returncode, stdout, stderr
src/python-common/ceph/rgw/rgwam_core.py:        cmd, returncode, stdout, stderr = self.mgr.tool_exec(self.prog, args)
src/python-common/ceph/rgw/rgwam_core.py:                      (returncode, cmd_str, stdout, stderr))
src/python-common/ceph/rgw/rgwam_core.py:            raise RGWAMCmdRunException(cmd_str, -returncode, stdout, stderr)
src/python-common/ceph/rgw/rgwam_core.py:        return (stdout, stderr)
src/python-common/ceph/rgw/rgwam_core.py:        return (retcode, stdout, stderr)
src/pybind/mgr/rgw/module.py:        return cmd, rc, stdout, stderr
src/pybind/mgr/selftest/module.py:            return HandleCommandResult(retval, stdout, stderr)
src/pybind/mgr/mgr_module.py:            return HandleCommandResult(retval, stdout, stderr)
src/pybind/mgr/mgr_module.py:            return result.returncode, stdout, stderr
src/pybind/mgr/dashboard/settings.py:                return value, stdout, stderr
src/test/pybind/test_ceph_argparse.py:        return ret, stdout, stderr
$ 

```
</details>

I'll stick to that and change `return (stdout, stderr)` to `return (0, stdout, stderr)`.